### ISSUE TITLE
Roles

### ISSUE THREAD

**Issue by sm2x** [openboxes/openboxes#450]
How do you create a new role that can do specific things like:
  - can access Analytics
  - can process Requisition  for Shipping

and how do you grant specific permissions to Managers to:
  - create Locations
  - create Users

**Comment by jmiranda** [comment]
Roles are hard-coded at the moment, so I'll need to create new roles for you. Can you elaborate on what you mean by "can access Analytics" and "can process Requisition for Shipping"?

As for granting those two permissions for the Manager role, I can either add those permissions to the Manager role or create a new role that sits between Manager and Admin. Can you let me know why Admin would not work in your case? In other words, what permissions does Admin provide that you don't want Manager to have?  

NOTE: There's a feature request for making roles / permissions more configurable, but I haven't had a chance to work on that yet. I'd like that have that included in 0.8.x though.

**Comment by sm2x** [comment]
Yes thats exactly I have in mind a role thats in between admin and manager, reason being, the admin will be given to IT people who are not directly involve in operations, but at the same time these specific functions are crucial to day to day activities.

Analytics menu only shows on admin account.

Requisitions will be coming from central office, to prepare shipping for a location (in this case a Customer location).  And they must not able to delete these requests but only cancel if they like.

**Comment by ewaterman** [comment]
This is quite an old thread so I'm opting to close this issue out (@sm2x feel free to re-open if needed).

However I think what we really need here is what Justin was aluding to: dynamically configurable roles. Likely this will come as a part of a Spring Security integration that we've been wanting to do so I won't create a specific task for that at the moment but we'd definitely benefit from dynamic roles.
